rfd/0056-cockroach-backend.md

```diff
@@ -0,0 +1,70 @@
+---
+authors: Jim Bishopp (jim@goteleport.com)
+state: draft
+---
+
+# RFD 56 - CockroachDB Backend
+
+## What
+
+A CockroachDB backend extends Teleport with the ability to store data
+in a self-hosted CockroachDB cluster.
+
+## Why
+
+Supporting additional backends such as CockroachDB reduces onboarding time
+for Teleport customers by allowing them to use existing infrastructure.
+
+## Scope
+
+This RFD focuses on implementing a backend for a self-hosted CockroachDB cluster.
+
+Requirements:
+
+- Prioritize self-hosted CockroachDB.
+- Implementation should support CockroachDB node failures by automatically
+  reconnecting to another node when a connected node becomes unavailable.
+- Design should consider extensibility to support other SQL databases 
+  such as PostgreSQL and MySQL.
+- Design should utilize a dialer/connector abstraction when connecting to
+  the database to allow the implementation to be extended to support SQL
+  databases hosted by a cloud provider.
+
+## Authentication
+
+The backend will require mutual authentication when connecting to the CockroachDB 
+cluster. This requires users to generate client certificates and configure
```

Comment:
I'm thinking, instead of using Cockroach's client certificates why not use the same approach as with Database Access and use Teleport-generated certificates (Cockroach database will need to be configured with Teleport's CA)? Then you'll be able to configure your Cockroach server same way for both database access and as a backend. We will need just an extra field like `user` in the configuration to indicate which user Teleport should use when connecting. What do you think?

Reply:
Isn't there a bit of chicken/egg.  HOw do you get a cert for your backend if it hasn't started to store somewhere?

Reply:
Good point 😄 Let's keep the current approach with Cockroach-signed certs for now, we can revisit later if needed.